test/legacy_test/test_affine_grid_function.py

```diff
@@ -78,11 +82,43 @@ def functional(self, place):
                     align_corners=self.align_corners,
                 )
         feed_dict = {"input": self.theta}
-        exe = base.Executor(place)
-        exe.run(start)
-        (y_np,) = exe.run(main, feed=feed_dict, fetch_list=[y_var])
+        exe = paddle.static.Executor(self.place)
+        (y_np,) = exe.run(
+            paddle.static.default_main_program(),
+            feed=feed_dict,
+            fetch_list=[y_var],
+        )
```

Comment:
```suggestion
            exe = paddle.static.Executor(place)
            (y_np,) = exe.run(
                paddle.static.default_main_program(),
                feed=feed_dict,
                fetch_list=[y_var],
            )
```
离开 `with paddle.static.program_guard(...,...)` 后会切换 `default_main_program` 的具体值。所以向执行器传入 paddle.static.default_main_program() 时需要保证其在 `program_guard` 的管理范围内，否则会找不到静态图构图时的变量

Reply:
我单独写了个，应该跟原始代码无关了吧